### PR TITLE
Fixes #28393 - consume eslint config from foreman-js

### PR DESCRIPTION
**Pull request by amirfefer**

### CODE REVIEW

**Comment by ekohl** [comment]
Nit: there's a typo in commit message (consune)

File: webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js
Comment (jeremylenz) [inline webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js:21]: Glad to see the Hooks eslint rules finally made it in :)

File: webpack/assets/javascripts/react_app/redux/index.js
Comment (jeremylenz) [inline webpack/assets/javascripts/react_app/redux/index.js]: Too bad the Hooks rule won't allow non-Hook functions whose names start with `use`.

How about `logReduxToConsole` or something more descriptive?

File: webpack/assets/javascripts/compute_resource/libvirt.js
Comment (jeremylenz) [inline webpack/assets/javascripts/compute_resource/libvirt.js:1]: Why disable all the jquery rules with magic comments, and not in eslint configuration?

File: webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js
Comment (jeremylenz) [reply webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js:21]: Looks like this same `getTableItems` dependency will also require a PR in `foreman-tasks`; not sure if that exists already
```
/home/vagrant/foreman/.vendor/ruby/2.5.0/gems/foreman-tasks-0.17.4/webpack/ForemanTasks/Components/TasksTable/TasksTable.js
  34:6  warning  React Hook useEffect has missing dependencies: 'getTableItems' and 'url'. Either include them or remove the dependency array. If 'getTableItems' changes too often, find the parent component that defines it and wrap that definition in useCallback  react-hooks/exhaustive-deps

✖ 1 problem (0 errors, 1 warning)
```

**Review by jeremylenz — COMMENTED** [review]
Tested and `npm run lint` lints successfully.  Thanks @amirfefer!  Small comments below

File: webpack/assets/javascripts/compute_resource/libvirt.js
Comment (amirfefer) [reply webpack/assets/javascripts/compute_resource/libvirt.js:1]: These comments are only for legacy js files that still use jquery.
With this approach, we would decrease the use of it over time

**Comment by amirfefer** [comment]
This PR also fixes the eslint issue on node 10

**Comment by sharvit** [comment]
Need a rebase

**Comment by mmoll** [comment]
[test foreman]

**Comment by mmoll** [comment]
rebased, integration tests fail :broken_heart:.

**Comment by tbrisker** [comment]
that's odd for mostly linting changes... [test foreman]

File: package.json
Comment (sharvit) [inline package.json]: Can be just `"lint": "tfm-lint"`

File: webpack/assets/javascripts/react_app/common/HOC.js
Comment (sharvit) [inline webpack/assets/javascripts/react_app/common/HOC.js:36]: @tbrisker @mmoll, this is more than a linting change

Comment (tbrisker) [reply webpack/assets/javascripts/react_app/common/HOC.js:36]: good point, i guess this is what is causing the test failures.

File: package.json
Comment (amirfefer) [reply package.json]: `tfm-lint: not found` :disappointed:

**Comment by mmoll** [comment]
[test foreman]

**Comment by mmoll** [comment]
[test foreman]

**Comment by tbrisker** [comment]
> This PR also fixes the eslint issue on node 10

Can you please add #28495 which is tracking that to the commit message so it's also closed when this is merged?

Also looks like integration tests are still failing.

**Comment by mmoll** [comment]
[test foreman]

**Review by tbrisker — DISMISSED** [review]
Looks like the last changed fixed the failing tests. Thanks @amirfefer ! waiting for tests to finish before merging.

File: webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js:21]: Should we add `getURIQuery(window.location.href)` to the array as well?
So it will fetch again when the URL changes.

**Comment by tbrisker** [comment]
We need to bump the vendor packaging to 3.9.3 as well to merge this, or our nightlies will fail to build due to the mismatching module version (currently we have 3.9.0 packaged)

**Comment by sharvit** [comment]
> We need to bump the vendor packaging to 3.9.3 as well to merge this, or our nightlies will fail to build due to the mismatching module version (currently we have 3.9.0 packaged)

Notice the latest vendor version is `3.12.0`, and there is a PR in core to upgrade:
https://github.com/theforeman/foreman/pull/7294

We have another PR in core wich use an unreleased version of the vendor:
https://github.com/theforeman/foreman/pull/7243
https://github.com/theforeman/foreman-js/pull/105
This PR will bring breaking changes so we will deploy the vendor as version `4.0.0`.

Can we wait with packaging to version `4.0.0`?
If not, it would be better to skip `3.9.3` and use `3.12.0`, then update to `4.0.0`

**Comment by tbrisker** [comment]
> Can we wait with packaging to version `4.0.0`?

we can, but we can't merge this until the packaging is updated. when will 4.0.0 be ready?

**Comment by sharvit** [comment]
> > Can we wait with packaging to version `4.0.0`?
> 
> we can, but we can't merge this until the packaging is updated. when will 4.0.0 be ready?

When https://github.com/theforeman/foreman-js/pull/105 will be ready. @amirfefer?

File: webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js
Comment (jeremylenz) [reply webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js:21]: @sharvit yeah probably

**Comment by amirfefer** [comment]
@tbrisker, @sharvit  - Rebased and ready :)

**Review by sharvit — APPROVED** [review]
Thanks @amirfefer, LGTM :+1:

**Review by tbrisker — APPROVED** [review]
Thanks @amirfefer and everyone!

**Pull request merged**
